garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,230 @@ local meta = FindMetaTable( "Vector" )
+Returns a copy of bisector of two vectors
+-----------------------------------------------------------]]
+function meta:GetBisected(...)
+	local v = Vector(self)
+	v:Bisect(...)
+	return v
+end
 
+--[[---------------------------------------------------------
+Convets the vector to resized version of itself
+-----------------------------------------------------------]]
+function meta:Resize(num)
+	self:Normalize()
+	self:Mul(num)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the resized vector
+-----------------------------------------------------------]]
+function meta:GetResized(...)
+	local v = Vector(self)
+	v:Resize(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to resized midpoint of itself
+-----------------------------------------------------------]]
+function meta:Midpoint(vec)
+	self:Add(vec)
+	self:Mul(0.5)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the midpoint vector
+-----------------------------------------------------------]]
+function meta:GetMidpoint(...)
+	local v = Vector(self)
+	v:Midpoint(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the zero point vector
+-----------------------------------------------------------]]
+function meta:GetZero(...)
+	local v = Vector(self)
+	v:Zero(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Calculates 3x3 determinant by using other two vectors as rows
+-----------------------------------------------------------]]
+function meta:Determinant(vec2, vec3)
+	local a, b, c = self:Unpack()
+	local d, e, f = vec2:Unpack()
+	local g, h, i = vec3:Unpack()
+	return ((a*e*i)+(b*f*g)+(d*h*c)-(g*e*c)-(h*f*a)-(d*b*i))
+end
+
+--[[---------------------------------------------------------
+Applies new basis to a vector by the three axises of an angle
+-----------------------------------------------------------]]
+function meta:Basis(ang)
+	local x = self:Dot(ang:Forward())
+	local y = self:Dot(ang:Right())
+	local z = self:Dot(ang:Up())
+	self:SetUnpacked(x, y, z)
+end
+
+--[[---------------------------------------------------------
+Applies new basis to a vector by the three axises of an angle
+-----------------------------------------------------------]]
+function meta:GetBasis(...)
+	local v = Vector(self)
+	v:Basis(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Projects the vector in to an axis defined by the second vector
+-----------------------------------------------------------]]
+function meta:Project(vec)
+	local vs = Vector(self)
+	self:Set(vec)
+	self:Normalize()
+	self:Mul(vs:Dot(self))
+end
+
+--[[---------------------------------------------------------
+Projects the vector in to an axis defined by the second vector
+-----------------------------------------------------------]]
+function meta:GetProject(...)

```

Comment:
Shouldn't this be GetProjected to match GetBisected and GetResized?

Reply:
Yes, it should. Fixed.

Reply:
It's not

Reply:
Ah.... Noticed the code bas is `Add` -> `GetAdd`

Reply:
Reverted. Also Added ray intersection fraction calculator

Reply:
On the other hand should I rename `GetAdd` `GetSub` to `GetAdded` `GetSubbed` .... :confused:

Reply:
I think all Get methods except for Midpoint and Add, Sub Mul, Div should end with ``ed`` since we currently have ``Unpack`` and ``SetUnpacked``. So ``GetProjected``, ``GetBisected``, ``GetResized``, and ``GetRotated``.

Reply:
Ah. Makes sense. Thank you! Updated !

Reply:
Whoops, forgot to mention to leave Basis alone, as well.

Reply:
Ok, Reverted! Thanks.